### PR TITLE
Remove deprecated jQuery event dispatching

### LINKED ISSUE #10356: Deprecation Warning in MainViewManager.js:1087 in master branch
```
 Core code should no longer be using $().on/off()! main.js:88
 jQuery.fn.initjquery-2.1.1.min.js:2 nMainViewManager.js:1087
_createPaneIfNecessaryMainViewManager.js:1619
_initialize MainViewManager.js:1677 
(anonymous function) AppInit.js:93 
_callHandler AppInit.js:113 
_dispatchReady brackets.js:471 
(anonymous function) jquery-2.1.1.min.js:2
n.Callbacks.j jquery-2.1.1.min.js:2 
n.Callbacks.k.fireWith jquery-2.1.1.min.js:2 
n.extend.Deferred.n.each.e.(anonymous function) PreferencesBase.js:294 
(anonymous function) jquery-2.1.1.min.js:2 
(anonymous function) jquery-2.1.1.min.js:2 
n.Callbacks.j jquery-2.1.1.min.js:2 
n.Callbacks.k.fireWith jquery-2.1.1.min.js:2
n.extend.Deferred.n.each.e.(anonymous function) PreferencesBase.js:190 
(anonymous function) File.js:125 
(anonymous function) AppshellFileSystem.js:378
(anonymous function) main.js:89
```

Code:

```js
            $(newPane).on("viewDestroy.mainView", function (e, view) {
                _removeFileFromMRU(newPane.id, view.getFile());
            });
```

cc @JeffryBooher

### PR DESCRIPTION
**Pull request by JeffryBooher**
Fixes #10356
CC: @redmunds

### CODE REVIEW

**Comment by redmunds** [comment]
This looks good on startup. But I selected a few different files, then switch to another project and see an exception: 

```js
 Uncaught TypeError: Cannot read property 'trigger' of undefined Pane.js:1125 
(anonymous function) Pane.js:1124 
Pane._reset MainViewManager.js:1385 
(anonymous function) lodash.js:9 
h lodash.js:23 
St MainViewManager.js:480
_forEach PaneOrPanesMainViewManager.js:1379 
_closeAll ProjectManager.js:836 
_load ProjectProjectManager.js:985 
(anonymous function) jquery-2.1.1.min.js:2
 j jquery-2.1.1.min.js:2 
k.add ProjectManager.js:982
open Projectmain.js:287 
(anonymous function) jquery-2.1.1.min.js:3 
n.event.dispatch jquery-2.1.1.min.js:3
```

**Comment by redmunds** [comment]
Also seeing 23 of 54 Main View unit tests fail.

**Comment by redmunds** [comment]
Done with initial review.

File: src/view/Pane.js [hunk 1]
```diff
@@ -509,7 +509,7 @@ define(function (require, exports, module) {
         
         // Destroy temporary views
         _.forEach(viewsToDestroy, function (view) {
-            $(this).triggerHandler("viewDestroy", view);
+            this.trigger("viewDestroy", view);
```
Comment (MarcelGerber) [inline src/view/Pane.js]: Use `self` here.

**Comment by JeffryBooher** [comment]
@redmunds fixed instances of `trigger` inside a callbacks.

**Comment by redmunds** [comment]
Merging.

**Pull request merged**